Makefile

```diff
@@ -26,7 +26,6 @@ test:
 build:
 	pip install -e .
 	python scripts/generate_from_specs.py
-	pre-commit run --all-files || true
```

Comment:
I'm not a huge fan of the recursion through `pre-commit` that I think is happening with these changes. 

So to make sure I understand, you added to the pre-commit config to run `make build` when the source files run, and thereby removed this part of `make build` where it runs `pre-commit`, which I recall was added to reformat the generated files in a consistent manner. But then in `scripts/generate_from_specs.py`, you added a call to run `pre-commit` but only on the output files:
```py
subprocess.call(["pre-commit", "run", "--files"] + output_files)
```

So technically the new `pre-commit` hook shouldn't run against the output files because they shouldn't match the regex, thereby not triggering `make build` again, but still the new hook would take `pre-commit` through `make`, `python`, then back to `pre-commit` again. I think I had brought up potentially having `python scripts/generate_from_specs.py` output the list of files for `pre-commit`.